src/include/daos_srv/dtx_srv.h

```diff
@@ -54,34 +46,52 @@ enum dtx_cos_list_types {
 /**
  * DAOS two-phase commit transaction handle in DRAM.
  */
-struct daos_tx_handle {
-	/** The identifier of the DTX */
-	struct daos_tx_id	 dth_xid;
-	/** The identifier of the object (shard) to be modified. */
-	daos_unit_oid_t		 dth_oid;
-	/* The dkey to be modified if applicable */
-	daos_key_t		*dth_dkey;
+struct dtx_handle {
+	union {
+		struct {
+			/** The identifier of the DTX */
+			struct dtx_id		dth_xid;
+			/** The identifier of the shard to be modified. */
+			daos_unit_oid_t		dth_oid;
+		};
+		struct dtx_entry		dth_dte;
```

Comment:
why do we have the union at here, they are the same?

Reply:
Some function, such as dtx_end() can directly use it as the parameter for dtx_commit/dtx_abort, then avoid new variable on stack.